Add Percona 8 support

mamedin opened the pull request:
This PR adds:

- MySQL 8 support
- Move "Ensure mysql service is up" task to configure files.
- Restart mysql when changing config file before adding users/databases.
- Remove `query_cache` variables for MySQL >= 8.0.
- Use `log_error_verbosity` instead of `log_warnings` when MySQL >= 8.0.
- Modify/Reorder secure.yml to make it clearer.
- Cast vars to int in my.cnf templates.
- Change the packages list format in yum annd apt tasks.
- Increase mininum ansible version to 2.5.

Connects to https://github.com/artefactual-labs/ansible-percona/issues/38

mamedin commented:
I included the changes proposed in PRs (not merged yet):

* https://github.com/artefactual-labs/ansible-percona/pull/36
* https://github.com/artefactual-labs/ansible-percona/pull/34

djjuhasz commented:
LGTM :) But I think @jraddaoui is going to be the best reviewer for this, since he did the dev work. :)  Thanks @mamedin .

jraddaoui commented on README.md:
```diff
@@ -5,7 +5,11 @@ Adapted from https://github.com/hpcloud-mon/ansible-percona
 ## Requirements
 
-None.
+Ansible >= 2.5
```
As we chatted, and < 2.9.

mamedin replied:
Done

jraddaoui commented on README.md:
```diff
@@ -5,7 +5,11 @@ Adapted from https://github.com/hpcloud-mon/ansible-percona
+
+For Mysql >= 8.0, the only authentication plugin currently supported is:
+
+`mysql_default_authentication_plugin: "mysql_native_password"`
```
I'd remove this lines. It's setting the auth. plugin to whatever is set in the related var and, if the var is not set, it will use `caching_sha2_password` by default.

jraddaoui commented on meta/main.yml:
```diff
@@ -3,7 +3,7 @@ galaxy_info:
   company: "Artefactual Systems"
   license: AGPLv3
-  min_ansible_version: 1.4
+  min_ansible_version: 2.5
```
Can we add a `max_ansible_version` in here?

mamedin commented:
Thanks Radda for suggestions :)

* Added `max_ansible_version` 
* Removed `mysql_default_authentication_plugin: "mysql_native_password"` from README.md
* Set `mysql_default_authentication_plugin: "mysql_native_password"` in defaults vars file.
* Added Percona Server 8 travis tests.

jraddaoui commented on templates/etc_my.cnf-centos.j2:
```diff
@@ -31,6 +31,9 @@ tmpdir      = /tmp
 {% if mysql_sql_mode is defined %}
 sql_mode={{ mysql_sql_mode }}
 {% endif %}
+{% if mysql_default_authentication_plugin is defined and mysql_version is version('5.7', '>=') %}
```
It'll always be defined now ;)

jraddaoui commented:
Hi @mamedin, it looks like the tests weren't triggered ... don't ask me why.

mamedin commented:
Reopen
